test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +92,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (command.isMutating()) {
+            assertEquals(addressBook, saveFile.load());
+        }
```

Comment:
Nice to see that you have updated the tests for this :) 

However, this change doesn’t help much because it does not confirm that the saving happens only for mutating commands. Tests will pass even if you don’t have this change. To be complete, you should have an else branch that confirms the file was not updated for non mutating commands (e.g. by checking the last updated time of the file).